networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/subnets/CommitteeSubnetSubscriptions.java

```diff
@@ -50,9 +51,8 @@ public synchronized void subscribeToSubnetId(final int subnetId) {
   }
 
   public synchronized void unsubscribeFromSubnetId(final int subnetId) {
-    final RequestedSubscription subscription = subnetIdToSubscription.remove(subnetId);
-    if (subscription != null) {
-      subscription.unsubscribe();
+    if (subnetIdToSubscription.containsKey(subnetId)) {
+      subnetIdToSubscription.remove(subnetId).unsubscribe();
```

Comment:
In my tests removing a value that doesn't exist returns `null` the same as a normal `HashMap`. As in this test passes:

```
  @Test
  void should() {
    final Int2ObjectOpenHashMap<Object> map = new Int2ObjectOpenHashMap<>();
    final Object result = map.remove(2);
    assertThat(result).isNull();
  }
```

And the Javadoc for `remove(int)` says:
```
Returns:
the old value, or the default return value if no value was present for the given key.
```

So we shouldn't need these additional `containsKey` checks.

Reply:
Yes, you are correct. I just reverted this change and it works fine. I don't know why I thought this. I must have been tracking down that other `iterator.remove()` behavior difference and got confused.